server/src/controllers/departmentController.ts

```
@@ -337,3 +337,49 @@ export const getExpenseDepartmentsByUser = async (
     next(error);
   }
 };
+
+export const getDataAnalysisDepartmentsByUser = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    if (!req.user?.id) {
+      res.status(401).json({ error: "Authenticated user not found" });
+      return;
+    }
+
+    const user = await User.findByPk(req.user.id, {
+      include: { model: Department, as: "departments" }
+    });
+
+    if (!user) {
+      res.status(200).json([]);
```

Here should return an error because user wasn't found, instead of 200, probably 404 or 401